contracts/DxMgnPool.sol

```diff
@@ -56,6 +58,9 @@ contract DxMgnPool is Ownable {
      * Public interface
      */
     function deposit(uint amount) public {
+        checkForStateUpdate();
```

Comment:
It's a little bit dangerous, that we have to remember to call this method now on every method that relies to have an up to date state. It was nicer, that it was implicitly calculated before, and didn't require a write to a local variable.

The alternative would be to still require access to the variable via a method (`currentState()`) and do the state update implicitly in there. But I'm not sure implicit state mutation in a getter is a good idea either.

Let's leave it like that.

Reply:
Yeah, I agree with your assessment